src/component/admin/auth/simple-auth.tsx

```diff
@@ -0,0 +1,96 @@
+import PageContent from '../../common/PageContent/PageContent';
+import AccessContext from '../../../contexts/AccessContext';
+import { ADMIN } from '../../providers/AccessProvider/permissions';
+import useAuthSettings from '../../../hooks/api/getters/useAuthSettings/useAuthSettings';
+import useAuthSettingsApi, {ISimpleAuthSettings } from '../../../hooks/api/actions/useAuthSettingsApi/useAuthSettingsApi';
+
+function SimpleAuth() {
+
+    const { config } = useAuthSettings('simple');
+    const [disabled, setDisabled] = useState<boolean>(false);
+    const [info, setInfo] = useState<string>();
+    const { updateSettings, errors } = useAuthSettingsApi<ISimpleAuthSettings>('simple')
+    const { hasAccess } = useContext(AccessContext);
+
+
+    useEffect(() => {
+        setDisabled(!!config.disabled);
+    }, [ config.disabled ]);
+
+    if (!hasAccess(ADMIN)) {
+        return (
+            <Alert severity="error">
+                You need to be a root admin to access this section.
+            </Alert>
+        );
+    }
+
+    const updateDisabled = () => {
+        setDisabled(!disabled);
+    };
+
+    
+    const onSubmit = async e => {
+        e.preventDefault();
+        
+        setInfo('...saving');
+        try {
+            const settings: ISimpleAuthSettings = { disabled };
+            await updateSettings(settings);
+            setInfo('Settings stored');
+            setTimeout(() => setInfo(''), 2000);
+        } catch (e) {
+            setInfo('');
+            setDisabled(config.disabled)
+        }
+        
+    };
+    return (
+        <PageContent headerContent=''>
+            <form onSubmit={onSubmit}>
+                <Grid container spacing={3}>
+                    <Grid item md={5}>
+                        <strong>Password based login</strong>
+                        <p>Allow users to login with username & password</p>
+                    </Grid>
+                    <Grid item md={6} style={{ padding: '20px' }}>
+                        <FormControlLabel
+                            control={
+                                <Switch
+                                    onChange={updateDisabled}
+                                    value={!disabled}
+                                    name="disabled"
+                                    checked={!disabled}
+                                />
+                            }
+                            label={!disabled ? 'Enabled' : 'Disabled'}
+                        />
+                    </Grid>
+                </Grid>
+                <Grid container spacing={3}>
+                    <Grid item md={12}>
+                        <Button
+                            variant="contained"
+                            color="primary"
+                            type="submit"
+                        >
+                            Save
+                        </Button>{' '}
+                        <small>{info}</small>
```

Comment:
Why not toggle this loading based on the free loading state you get from your API hook instead of using an internal state?

Reply:
because I am not fluent in the useSwr hooks yet, and it was partially based on the existing code. Will take a stab at this!